### PR TITLE
[tlv] add new tlvs for network diagnostic

### PR DESCRIPTION
**Pull request by ZhangLe2016**
libotcommissioner.jar, a versatile Java library, provides support for wide range of TMF message transactions, including MeshCop protocols and diagnostic TLVs. This PR focuses on integrating all the diagnostic TLVs, as defined in the spec `10.11.3` into the library's TLV modulei

### CODE REVIEW

**Review by wgtdkp — APPROVED** [review]
LGTM :+1:

**Review by lanyuwen — APPROVED** [review]
LGTM and thanks!

File: src/library/tlv.cpp [hunk 1]
```diff
@@ -220,6 +220,73 @@ bool Tlv::IsValid() const
...
+        case Type::kNetworkDiagVendorName:
+            return length >= 4;
+        case Type::kNetworkDiagVendorModel:
+            return length >= 4;
+        case Type::kNetworkDiagVendorSWVersion:
+            return length >= 2;
+        case Type::kNetworkDiagChild:
+            return length >= 43;
+        case Type::kNetworkDiagChildIpv6Address:
+            return (length % 16 == 0) && (length / 16 >= 1 && length / 16 <= 15);
```
[the remark this answers is not in the document]
Comment (ZhangLe2016) [reply src/library/tlv.cpp:277]: Thanks, changed.

**Review by abtink — APPROVED** [review]
Thanks.

**Review by jwhui — APPROVED** [review]

**Pull request merged**